ls -Lt produces different results than GNU ls when given absolute path

I'm trying to compile curl, and it's failing on this bit of the configure script:

```bash
# Do 'set' in a subshell so we don't clobber the current shell's
# arguments.  Must try -L first in case configure is actually a
# symlink; some systems play weird games with the mod time of symlinks
# (eg FreeBSD returns the mod time of the symlink's containing
# directory).
if (
   set -x
   pwd
   am_has_slept=no
   for am_try in 1 2; do
     echo "timestamp, slept: $am_has_slept" > conftest.file
     set X `ls -Lt "$srcdir/configure" conftest.file 2> /dev/null`
     if test "$*" = "X"; then
	# -L didn't work.
	set X `ls -t "$srcdir/configure" conftest.file`
     fi
     if test "$*" != "X $srcdir/configure conftest.file" \
	&& test "$*" != "X conftest.file $srcdir/configure"; then

	# If neither matched, then we have a broken ls.  This can happen
	# if, for instance, CONFIG_SHELL is bash and it inherits a
	# broken ls alias from the environment.  This has actually
	# happened.  Such a system could not be considered "sane".
	as_fn_error $? "ls -t appears to fail.  Make sure there is not a broken
  alias in your environment" "$LINENO" 5
     fi
     if test "$2" = conftest.file || test $am_try -eq 2; then
       break
     fi
     # Just in case.
     sleep 1
     am_has_slept=yes
   done
   test "$2" = conftest.file
   )
then
   # Ok.
   :
else
   as_fn_error $? "newly created file is older than distributed files!
Check your system clock" "$LINENO" 5
fi
```

Basically, what's happening there is it's trying to create a file, compare the timestamp of that file against the `./configure` file, and ensure that `ls` is returning the files in timestamp order.

However, this fails because Rust coreutils ls, when given:

```
ls -Lt /absolute/path/to/configure conftest.file
```

returns:

```
conftest.file configure
```

whereas, GNU ls returns:

```
conftest.file /absolute/path/to/configure
```

Since the configure script is trying to do an exact match on the returned value, this causes the configure script to fail - it incorrectly says the newly created file is older than distributed files, but the error is actually about the output from ls. 

```
checking whether build environment is sane... configure: error: ls -t appears to fail.  Make sure there is not a broken
  alias in your environment
configure: error: newly created file is older than distributed files!
Check your system clock
```

--- a/src/uu/ls/src/ls.rs
+++ b/src/uu/ls/src/ls.rs
@@ -1110,7 +1110,7 @@ struct PathData {
     md: OnceCell<Option<Metadata>>,
     ft: OnceCell<Option<FileType>>,
     // Name of the file - will be empty for . or ..
-    file_name: String,
+    display_name: String,
     // PathBuf that all above data corresponds to
     p_buf: PathBuf,
     must_dereference: bool,
@@ -1126,14 +1126,18 @@ impl PathData {
     ) -> Self {
         // We cannot use `Path::ends_with` or `Path::Components`, because they remove occurrences of '.'
         // For '..', the filename is None
-        let name = if let Some(name) = file_name {
+        let display_name = if let Some(name) = file_name {
             name
         } else {
-            p_buf
-                .file_name()
-                .unwrap_or_else(|| p_buf.iter().next_back().unwrap())
-                .to_string_lossy()
-                .into_owned()
+            let display_osstr = if command_line {
+                p_buf.as_os_str()
+            } else {
+                p_buf
+                    .file_name()
+                    .unwrap_or_else(|| p_buf.iter().next_back().unwrap())
+            };
+
+            display_osstr.to_string_lossy().into_owned()
         };
         let must_dereference = match &config.dereference {
             Dereference::All => true,
@@ -1159,7 +1163,7 @@ impl PathData {
         Self {
             md: OnceCell::new(),
             ft,
-            file_name: name,
+            display_name,
             p_buf,
             must_dereference,
         }
@@ -1243,7 +1247,7 @@ fn sort_entries(entries: &mut Vec<PathData>, config: &Config) {
             entries.sort_by_key(|k| Reverse(k.md().as_ref().map(|md| md.len()).unwrap_or(0)))
         }
         // The default sort in GNU ls is case insensitive
-        Sort::Name => entries.sort_by(|a, b| a.file_name.cmp(&b.file_name)),
+        Sort::Name => entries.sort_by(|a, b| a.display_name.cmp(&b.display_name)),
         Sort::Version => entries.sort_by(|a, b| version_cmp::version_cmp(&a.p_buf, &b.p_buf)),
         Sort::Extension => entries.sort_by(|a, b| {
             a.p_buf
@@ -1719,7 +1723,7 @@ fn classify_file(path: &PathData) -> Option<char> {
 }
 
 fn display_file_name(path: &PathData, config: &Config) -> Option<Cell> {
-    let mut name = escape_name(&path.file_name, &config.quoting_style);
+    let mut name = escape_name(&path.display_name, &config.quoting_style);
 
     #[cfg(unix)]
     {

--- a/tests/by-util/test_ls.rs
+++ b/tests/by-util/test_ls.rs
@@ -1967,8 +1967,6 @@ fn test_ls_sort_extension() {
     );
 }
 
-// This tests for the open issue described in #2223
-#[cfg_attr(not(feature = "test_unimplemented"), ignore)]
 #[test]
 fn test_ls_path() {
     let scene = TestScenario::new(util_name!());
@@ -1987,13 +1985,17 @@ fn test_ls_path() {
     scene.ucmd().arg(path).run().stdout_is(expected_stdout);
 
     let expected_stdout = &format!("./{}\n", path);
-    scene.ucmd().arg(path).run().stdout_is(expected_stdout);
+    scene
+        .ucmd()
+        .arg(format!("./{}", path))
+        .run()
+        .stdout_is(expected_stdout);
 
-    let abs_path = format!("{}/{}\n", at.as_string(), path);
-    println!(":{}", abs_path);
-    scene.ucmd().arg(&abs_path).run().stdout_is(&abs_path);
+    let abs_path = format!("{}/{}", at.as_string(), path);
+    let expected_stdout = &format!("{}\n", abs_path);
+    scene.ucmd().arg(&abs_path).run().stdout_is(expected_stdout);
 
-    let expected_stdout = &format!("{}\n{}\n", file1, path);
+    let expected_stdout = &format!("{}\n{}\n", path, file1);
     scene
         .ucmd()
         .arg(file1)

--- a/tests/by-util/test_ls.rs
+++ b/tests/by-util/test_ls.rs
@@ -1992,10 +1992,18 @@ fn test_ls_path() {
         .stdout_is(expected_stdout);
 
     let abs_path = format!("{}/{}", at.as_string(), path);
-    let expected_stdout = &format!("{}\n", abs_path);
+    let expected_stdout = if cfg!(windows) {
+        format!("\'{}\'\n", abs_path)
+    } else {
+        format!("{}\n", abs_path)
+    };
     scene.ucmd().arg(&abs_path).run().stdout_is(expected_stdout);
 
-    let expected_stdout = &format!("{}\n{}\n", path, file1);
+    let expected_stdout = if cfg!(windows) {
+        format!("{}  {}\n", path, file1)
+    } else {
+        format!("{}\n{}\n", path, file1)
+    };
     scene
         .ucmd()
         .arg(file1)

